AWSCognitoIdentityProvider/AWSCognitoIdentityUser.m

```diff
@@ -1488,6 +1488,28 @@ -(BOOL) isSignedIn {
     return refreshToken != nil;
 }
 
+- (AWSTask<AWSCognitoIdentityProviderRevokeTokenResponse *> *) revokeToken {
+    __block NSString * keyChainNamespace = [self keyChainNamespaceClientId];
+    NSString * accessTokenKey = [self keyChainKey:keyChainNamespace key:AWSCognitoIdentityUserAccessToken];
+    NSString * accessTokenString = self.pool.keychain[accessTokenKey];
+    AWSCognitoIdentityUserSessionToken * accessToken = [[AWSCognitoIdentityUserSessionToken alloc] initWithToken:accessTokenString];
+    if ([accessToken.tokenClaims objectForKey:@"origin_jti"] == nil) {
+        AWSDDLogVerbose(@"Access Token does not contain `origin_jti` claim. Skip revoking tokens.");
+        return [AWSTask taskWithResult:nil];
```

Comment:
Should this be an error? It would be better to send either a success object or an error back from this api. With the current code we are passing three types of response:
1. nil (if origin_jti not present)
2. AWSCognitoIdentityProviderRevokeTokenResponse (if api succeeded)
3. Error (if api failed)

Reply:
this is client side validation for backwards compatible tokens which do not have the revocation feature enabled, it  is also used to avoid an extra API call to Cognito. If the token doesn't contain "origin_jti" and we continue to call Cognito with revoke token, we'll get an error response indicating that the feature is not enabled. If the whole purpose is to avoid the network call then we can return the same response as if we made the service call.

Reply:
we can also separate out the API into "shouldRevoke -> Bool" and "revokeToken", which would reduce the response down to AWSCognitoIdentityProviderRevokeTokenResponse and Error.

Reply:
so we can 
1. keep it as is, return nil on success which indiciates that this revokeToken API did no-op
2. keep as is but return an error, to closely align with what the service would return
3. separate out the APis into "isSessionRevocable" and "revokeToken". this dumbs down this revokeToken as just an API call directly to the LLC revoke token API

Reply:
as discusssed, our preference is for 3, and to see if we are happy with the calling code in awsmobileclient